Catalog.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System.Net.Http.Json;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+namespace Lekcia11_cvicenie
+
+{
+    internal class Catalog
+    {
+        private List<Product> _products;
+
+        public Catalog() 
+        { 
+            _products = new List<Product>();
+        }
+
+        public void AddProduct(Product product)
+        {
+                _products.Add(product);
+        }
+
+        public List<string> SerializeProducts()
+        {
+            return _products.Select(product => JsonSerializer.Serialize(product)).ToList();
+        }
+
+        public void DeserializeProducts(List<string> jsonProducts)
+        {
+            foreach (var json in jsonProducts)
+            {
+                try
+                {
+                    var product = JsonSerializer.Deserialize<Product>(json);
+                    if (product != null)
+                    {
+                        _products.Add(product);
+                    }
```

Comment:
tenhle cely if by mohl byt idealne v metode `AddProduct` a odsud ji jen zavolat
```suggestion
                    AddProduct(product);
```